src/riak_kv_vnode.erl

```diff
@@ -1111,14 +1111,23 @@ handle_command({rebuild_complete, trees, _ST}, _Sender, State) ->
             {noreply, State#state{tictac_rebuilding = false}}
     end;
 
-handle_command({exchange_complete, {_EndState, DeltaCount}, ST},
+handle_command({exchange_complete, {EndState, DeltaCount}, ST},
                                                     _Sender, State) ->
     %% Record how many deltas were seen in the exchange
     %% Revert the skip_count to 0 so that exchanges can be made at the next
     %% prompt.
     XC = State#state.tictac_exchangecount + 1,
     DC = State#state.tictac_deltacount + DeltaCount,
     XT = State#state.tictac_exchangetime + timer:now_diff(os:timestamp(), ST),
+    case EndState of
+        PositiveState
+            when PositiveState == root_compare;
+                 PositiveState == branch_compare ->
+            ok;
+        UnwelcomeState ->
+            lager:info("Tictac AAE exchange for partition=~w pending_state=~w",
+                        [State#state.idx, UnwelcomeState])
+    end,
```

Comment:
Not sure you like list comprehensions, but for such a side-effect: do this if Condition, they are really nice.
The above code would become a one-liner:
```
     [ lager:info("Tictac AAE exchange for partition=~w pending_state=~w",
       [State#state.idx, EndState]) || EndState /= root_compare andalso EndState /= branch_compare].
```
Instead of `andelse` you can use `,` or `and`.
But you may also absolutely dislike it :-).

Reply:
I'm more concerned that I've leaked some knowledge of an atom in kv_index_tictactree into riak_kv.  I think really I should have a `aae_exchange:positive_outcomes/0` exported, so that riak_kv doesn't need to know about what these mean, and is decoupled from new positive outcomes being added.

I will look at this.

Reply:
Having fiddled, I'm only making matters worse -swapping one bit of special knowledge for another.  will leave as-is